OpenRA.Game/World.cs

```diff
@@ -26,7 +26,7 @@ public class World
 	{
 		static readonly Func<int, int, bool> FalsePredicate = (u, v) => false;
 		internal readonly TraitDictionary TraitDict = new TraitDictionary();
-		readonly HashSet<Actor> actors = new HashSet<Actor>();
+		readonly List<Actor> actors = new List<Actor>();
```

Comment:
In the SyncHash method, the actors and effects in enumerated to generate the hash, and the order of enumeration is important in this method. Since HashSet has no defined order, this is undefined behaviour.

We need collections with a defined order, so I just went with List to match the existing effects collection.

Thinking about it though, SortedSet makes more sense - might change that.